pages/weight.js

```diff
@@ -23,39 +25,70 @@ export async function getServerSideProps(context) {
 }
 
 export default function WeightPage({ days }) {
-  const filteredDays = days.filter((day) => Boolean(day.weights.length > 0));
+  const { data: session } = useSession();
+  if (session) {
+    const filteredDays = days.filter((day) => Boolean(day.weights.length > 0));
 
-  const ascendingFilteredDays = Array.from(filteredDays).reverse();
+    const ascendingFilteredDays = Array.from(filteredDays).reverse();
```

Comment:
And the same applies here again (reverse/sort).

Reply:
same as above